include/kernels/clover_deriv.cuh

```diff
@@ -33,9 +33,11 @@ namespace quda
     }
   };
 
-  template <typename Link, typename Force, typename Arg>
-  __device__ __host__ void computeForce(Force &force_total, const Arg &arg, int xIndex, int parity, int mu, int nu)
+  using computeForceOps = KernelOps<thread_array<int, 4>>;
```

Comment:
I think we can get rid of this extra abstraction and just have `CloverDerivative` inherit directly from `KernelOps<thread_array<int, 4>>`